core/transaction/transaction.go

```diff
@@ -101,7 +101,9 @@ func (tx *Transaction) SerializeUnsigned(w io.Writer) error {
 	}
 	tx.Payload.Serialize(w)
 	//nonce
-	serialization.WriteVarUint(w, tx.Nonce)
+	if len(tx.BalanceInputs) > 0 {
```

Comment:
as the first plan of <Legal chain>  , Antshare and DNA will run together for a period. i.e : the transaction will be send to the 2 chain.   So the request is transaction hash should be the same.  as antshare with no balance ,so don't have `nonce`.   after discussed with lijun, we think it can be removed out of serialization when with no balance inputs. but this is not the final solution. but just make it work for v0.6.